File: ASSIGNMENT9/python/Q.7.py
def find_pairs_with_difference(arr, diff):
    seen = set(arr)
    pairs = [(num, num - diff) for num in arr if (num - diff) in seen]
    return pairs

arr = [5, 20, 3, 2, 50, 80]
diff = 78
print("Pairs with difference {}: {}".format(diff, find_pairs_with_difference(arr, diff)))

File: ASSIGNMENT9/c/Q.7.c
#include <stdio.h>

void findPairsWithDifference(int arr[], int n, int diff) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = i + 1; j < n; j++) {
            if (abs(arr[i] - arr[j]) == diff) {
                printf("Pair found (%d, %d)\n", arr[i], arr[j]);
            }
        }
    }
}

int main() {
    int arr[] = {1, 5, 3, 4, 2};
    int diff = 2;
    int n = sizeof(arr) / sizeof(arr[0]);
    findPairsWithDifference(arr, n, diff);
    return 0;
}
